### PR TITLE
Add BindingsOfClosure helper

### PR DESCRIPTION
**Pull request by fingolfin**
Partially addresses #3569.

Well, or maybe it addresses it completely, though one still might want to enhance the output of `Display(func)` if there are bindings? Somebody else could do that based on this PR, though.

I also guess it would be nice to have documentation for `BindingsOfClosure`...

### CODE REVIEW

File: src/vars.c [hunk 1]
```diff
@@ -2008,13 +2008,16 @@ static Obj FuncParentLVars(Obj self, Obj lvars)
...
+      RequireArgument("ParentLVars", lvars, "must be an lvars");
+  }
   Obj contents = NEW_PREC(0);
   Obj func = FUNC_LVARS(lvars);
   Obj nams = NAMS_FUNC(func);
   UInt len = (SIZE_BAG(lvars) - 2*sizeof(Obj) - sizeof(UInt))/sizeof(Obj);
   Obj values = NEW_PLIST_IMM(T_PLIST, len);
   if (lvars == STATE(BottomLVars))
-    return False;
+    return Fail;
```
Comment (PaulaHaehndel) [inline src/vars.c]: I assume then the test (what travis tests) should be changed accordingly.

Comment (fingolfin) [reply src/vars.c]: Updated

**Comment by wilfwilson** [comment]
@fingolfin Do you intend to document `BindingsOfClosure` in this PR?

**Comment by fingolfin** [comment]
I don't have the time and energy to work on documentation for `BindingsOfClosure` right now. I am not sure we'd want to document it as such globally, either... so there are multiple paths forward:
1. We merge this now, then for a few people `BindingsOfClosure` already is useful; and update #3569 to mention it, suggesting that somebody else submit a PR to enhance `Display` for closures using it (and also that they could consider documenting `BindingsOfClosure` as well).
2. We merge it until somebody has time and energy to write some documentation, and tack it onto this PR.

I think 1. is better because it means this code won't bitrot, and people can benefit from it already (even if the number of people doing so would be small, as it is not documented). But either is fine by me.

**Comment by wilfwilson** [comment]
I'm happy for us to do 1. I thought that _perhaps_ your sentence
> I also guess it would be nice to have documentation for `BindingsOfClosure`...

suggested that you were maybe thinking of creating documentation, which is why I asked about it.

**Pull request merged**